Sources/WalletConnectSign/Services/HistoryService.swift

```diff
@@ -17,15 +27,33 @@ final class HistoryService {
     func getPendingRequests(topic: String) -> [Request] {
         return getPendingRequests().filter { $0.topic == topic }
     }
-
-    public func getSessionRequest(id: RPCID) -> Request? {
-        guard let record = history.get(recordId: id) else { return nil }
-        return mapRequestRecord(record)
+    
+    func getPendingProposals() -> [Session.Proposal] {
+        let pendingHistory = history.getPending()
+        
+        let requestSubscriptionPayloads = pendingHistory
+            .compactMap { record -> RequestSubscriptionPayload<SessionType.ProposeParams>? in
+                guard let proposalParams = mapProposeParams(record) else {
+                    return nil
+                }
+                return RequestSubscriptionPayload(id: record.id, topic: record.topic, request: proposalParams, publishedAt: Date())
+            }
+        
+        requestSubscriptionPayloads.forEach {
+            let proposal = $0.request
+            proposalPayloadsStore.set($0, forKey: proposal.proposer.publicKey)
```

Comment:
why we cannot remove proposalPayloadsStore and depend on rpc history only? 

If big array of expired proposals the only reason I think it's fine. We can fix it when implement expiration logic for proposals